File: designSpreadsheet.py
class Spreadsheet:
    def __init__(self, rows: int):
        self.rows = rows
        self.grid = [[0] * 26 for _ in range(rows)]

    def _parse_cell(self, cell: str):
        col = ord(cell[0]) - ord('A')
        row = int(cell[1:]) - 1
        return row, col

    def setCell(self, cell: str, value: int) -> None:
        r, c = self._parse_cell(cell)
        self.grid[r][c] = value

    def resetCell(self, cell: str) -> None:
        r, c = self._parse_cell(cell)
        self.grid[r][c] = 0

    def getValue(self, formula: str) -> int:
        parts = formula[1:].split('+')  # remove '=' and split
        total = 0
        for part in parts:
            if part[0].isalpha():  # cell reference
                r, c = self._parse_cell(part)
                total += self.grid[r][c]
            else:  # integer
                total += int(part)
        return total

File: designSpreadsheet.go
package main

import (
    "strconv"
    "unicode"
)

type Spreadsheet struct {
    rows int
    grid [][]int
}

func Constructor(rows int) Spreadsheet {
    grid := make([][]int, rows)
    for i := range grid {
        grid[i] = make([]int, 26)
    }
    return Spreadsheet{rows: rows, grid: grid}
}

func (s *Spreadsheet) parseCell(cell string) (int, int) {
    col := int(cell[0] - 'A')
    row, _ := strconv.Atoi(cell[1:])
    return row - 1, col
}

func (s *Spreadsheet) SetCell(cell string, value int) {
    r, c := s.parseCell(cell)
    s.grid[r][c] = value
}

func (s *Spreadsheet) ResetCell(cell string) {
    r, c := s.parseCell(cell)
    s.grid[r][c] = 0
}

func (s *Spreadsheet) GetValue(formula string) int {
    parts := []rune(formula[1:]) // skip '='
    splitIdx := -1
    for i, ch := range parts {
        if ch == '+' {
            splitIdx = i
            break
        }
    }
    left := string(parts[:splitIdx])
    right := string(parts[splitIdx+1:])

    return s.evalToken(left) + s.evalToken(right)
}

func (s *Spreadsheet) evalToken(token string) int {
    if len(token) > 0 && unicode.IsLetter(rune(token[0])) {
        r, c := s.parseCell(token)
        return s.grid[r][c]
    }
    val, _ := strconv.Atoi(token)
    return val
}
